src/Eagle.ts

```diff
@@ -2585,7 +2586,37 @@ export class Eagle {
         return true;
     }
 
-   
+    spinCollapseIcon = (item:any, e:JQueryEventObject) => {
```

Comment:
There are just a few "style" issues here. The code is fine, it just doesn't exactly fit with the existing style

1. The variable name "CollapseTarget" begins with a capital letter, I tend to use a lower-case first letter in the camel case for this Typescript
2. Most lines do not end with a semi-colon, which seems to be valid Typescript, but I think we should keep it consistent.
3. The indenting of line 2604 has one tab too many.
4. Two line gap on lines 2597-8 should just be one line
5. The name of the new class you have defined "ClosedIcon" should start with a lower-case letter
6. Lines 2604-2608 and 2612-2616 seem to be repeated. I think it would be better to remove this code from both the if and else blocks, and just have one copy outside (after) the if-else